### ISSUE TITLE
iotedge 'check' command produces "error" message for devices provisioned with DPS

### ISSUE THREAD

**Issue by toolboc** [Azure/iotedge#2500]
## Expected Behavior
The check script should conditionally test according to the configuration type, i.e. (Manual, DPS TPM, DPS Symmetric key).  If there is no IoT Hub connection (in the case of DPS), do not execute those tests to avoid both the warnings and error message that are produced today.

## Current Behavior
Output of `iotedge check` produces "error" message for devices provisioned with DPS.  Warnings might be acceptable but the error message is cause for concern.  

## Steps to Reproduce
1. Provision an IoTEdge device using DPS
2. Run `iotedge check`
3. Notice error message related to missing manual configuration

### Output of `iotedge check`
```
Configuration checks
--------------------
√ config.yaml is well-formed - OK
× config.yaml has well-formed connection string - Error
    Device is not using manual provisioning, so Azure IoT Hub hostname needs to be specified with --iothub-hostname
√ container engine is installed and functional - OK
√ config.yaml has correct hostname - OK
√ config.yaml has correct URIs for daemon mgmt endpoint - OK
√ latest security daemon - OK
√ host time is close to real time - OK
√ container time is close to host time - OK
√ DNS server - OK
‼ production readiness: certificates - Warning
    Device is using self-signed, automatically generated certs.
    Please see https://aka.ms/iotedge-prod-checklist-certs for best practices.
√ production readiness: certificates expiry - OK
‼ production readiness: container engine - Warning
    Device is not using a production-supported container engine (moby-engine).
    Please see https://aka.ms/iotedge-prod-checklist-moby for details.
‼ production readiness: logs policy - Warning
    Container engine is not configured to rotate module logs which may cause it run out of disk space.
    Please see https://aka.ms/iotedge-prod-checklist-logs for best practices.
    You can ignore this warning if you are setting log policy per module in the Edge deployment.

Connectivity checks
-------------------
√ host can connect to and perform TLS handshake with DPS endpoint - OK
‼ host can connect to and perform TLS handshake with IoT Hub AMQP port - Warning
    skipping because of previous failures
‼ host can connect to and perform TLS handshake with IoT Hub HTTPS / WebSockets port - Warning
    skipping because of previous failures
‼ host can connect to and perform TLS handshake with IoT Hub MQTT port - Warning
    skipping because of previous failures
‼ container on the default network can connect to IoT Hub AMQP port - Warning
    skipping because of previous failures
‼ container on the default network can connect to IoT Hub HTTPS / WebSockets port - Warning
    skipping because of previous failures
‼ container on the default network can connect to IoT Hub MQTT port - Warning
    skipping because of previous failures
‼ container on the IoT Edge module network can connect to IoT Hub AMQP port - Warning
    skipping because of previous failures
‼ container on the IoT Edge module network can connect to IoT Hub HTTPS / WebSockets port - Warning
    skipping because of previous failures
‼ container on the IoT Edge module network can connect to IoT Hub MQTT port - Warning
    skipping because of previous failures
√ Edge Hub can bind to ports on host - OK

11 check(s) succeeded.
3 check(s) raised warnings.
1 check(s) raised errors.
9 check(s) were skipped due to errors from other checks.
```

![image](https://user-images.githubusercontent.com/2018336/74184432-6a012580-4c0c-11ea-843d-0f54e7e9cbd8.png)

</details>

### Device Information
* NVIDIA L4T 32.3.1
* arm64
* Linux containers

### Runtime Versions
* iotedge 1.0.8 
* Edge Agent 1.0
* Edge Hub 1.0 
* Docker 18.09.7

**Comment by veyalla** [comment]
Thanks for report, this is being addressed in 1.0.9 release.

**Comment by arsing** [comment]
Duplicate of https://github.com/Azure/iotedge/issues/2313
